content/docs/user-guide/testing/best-practices/_index.md

```diff
@@ -0,0 +1,365 @@
+
+* Unit tests should account for the vast majority of the total number of tests.
+  * The most efficient to write, execute, and maintain.
+  * First line of automated defense against newly introduced bugs.
+  * Often referred to as the base of a "Testing Pyramid"
+* Integration tests should account for most of the remaining automated tests.
+  * Integration often catch the most variety of bugs, but also require significantly more effort to write and maintain.
+  * UI tests should extremely limited. Most integration tests can be written just below the UI layer, to verify the core functionality of the systems.
+* System tests should account for a few tests.
+  * Only the few most critical few workflows should be included.
+  * Should not aim to be exhaustive.
+* Performance metrics and other specialized tests should be added only when specific need is identified.
+
+## Where Do I Write New Tests?
+
+Low-level tests such as Unit Tests and Micro-Benchmarks should live adjacent to the code being tested, in a folder named `/Tests/`. Test-specific dependencies such as mock interfaces should exist in a similar location, such as `/Mocks/`. When modifying code for an existing feature, look for tests which already exist.
+
+Higher-level tests have more dependencies, and should live in a directory common to their set of dependencies. This is often an [O3DE Project](/docs/user-guide/project-config/).
+
+For more information on how to write and register automated tests, refer to the [Testing Quickstart Guide](/docs/user-guide/testing/getting-started/).
+
+## Test Names
+
+Tests should have unique, obvious names to help document what they perform. One highly recommended pattern is the [Osherove Naming](http://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html) structure of `<UnitOfWork>_<StateUnderTest>_<ExpectedBehavior>`. This maps to the "Arrange, Act, Assert" test structure, and keeps test names readable at a glance:
+
+1. UnitOfWork: The critical interaction verified. (Act)
+   * Typically named for single action, often including the name of the target function.
+   * Easy to read, since it comes first.
+2. StateUnderTest: Unique configuration steps. (Arrange)
+   * Describes what makes the test vary from "normal" conditions in similar tests.
+     * When nothing seems special use a placeholder such as `Default` or `HappyPath`, or omit this.
+3. ExpectedBehavior: The expected outcome of the test. (Assert)
+   * Often describes the type of assertion being made, or highlights the most important of multiple asserted values.
+   * Easy to read, since it comes last.
+
+An alternative ways to think about this pattern is `<WhatIsTested>_<NotableConfigurationStep>_<ImportantAssertion>`
+
+There are two main reasons to use this naming method:
+
+1. When a test fails, a human can look at a report summary and quickly form an idea of what went wrong.
```

Comment:
```suggestion
1. When a test fails, a human can look at a report summary and quickly understand what went wrong.
```

Reply:
I think this loses a lot of color about not fully understanding, but I'm okay accepting the recommendation.